scripts/zfs-helpers.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+#
+# This script is designed to facilitate in-tree development and testing
+# by installing symlinks on your system which refer to in-tree helper
+# utilities.  These helper utilities must be installed to in order to
+# exercise all ZFS functionality.  By using symbolic links and keeping
+# the scripts in-tree during development they can be easily modified
+# and those changes tracked.
+#
+# Use the following configuration option to override the installation
+# paths for these scripts.  The correct path is automatically set for
+# most distributions but you can optionally set it for your environment.
+#
+#   --with-mounthelperdir=DIR
+#                           install mount.zfs in dir [[/sbin]]
+#   --with-udevdir=DIR      install udev helpers [default=check]
+#   --with-udevruledir=DIR  install udev rules [[UDEVDIR/rules.d]]
+#
+
+basedir="$(dirname $0)"
+
+SCRIPT_COMMON=common.sh
+if [ -f "${basedir}/${SCRIPT_COMMON}" ]; then
+. "${basedir}/${SCRIPT_COMMON}"
+else
+echo "Missing helper script ${SCRIPT_COMMON}" && exit 1
+fi
+
+PROG=zfs-helpers.sh
+DRYRUN=
+INSTALL=
+REMOVE=
+VERBOSE=
+
+usage() {
+cat << EOF
+USAGE:
+$0 [dhirv]
+
+DESCRIPTION:
+	Install/remove the ZFS helper utilities.
+
+OPTIONS:
+	-d      Dry run
+	-h      Show this message
+	-i      Install the helper utilties
+	-r      Remove the helper utilities
+	-v      Verbose
+
+$0 -iv
+$0 -r
+
+EOF
+}
+
+while getopts 'hdirv' OPTION; do
+	case $OPTION in
+	h)
+		usage
+		exit 1
+		;;
+	d)
+		DRYRUN=1
+		;;
+	i)
+		INSTALL=1
+		;;
+	r)
+		REMOVE=1
+		;;
+	v)
+		VERBOSE=1
+		;;
+	?)
+		usage
+		exit
+		;;
+	esac
+done
+
+if [ "${INSTALL}" -a "${REMOVE}" ]; then
+	usage
+	die "Specify -i or -r but not both"
+fi
+
+if [ ! "${INSTALL}" -a ! "${REMOVE}" ]; then
+	usage
+	die "Either -i or -r must be specified"
+fi
+
+if [ $(id -u) != 0 ]; then
+	die "Must run as root"
+fi
+
+if [ "$VERBOSE" ]; then
+	echo "--- Configuration ---"
+	echo "udevdir:          $udevdir"
+	echo "udevruledir:      $udevruledir"
+	echo "mounthelperdir:   $mounthelperdir"
+	echo "DRYRUN:           $DRYRUN"
+	echo
+fi
+
+install() {
+	local src=$1
+	local dst=$2
+
```

Comment:
How about testing for the existence of $src?
If the name of one of these files changes in the future, a warning from this script would be helpful.